tsdb/querier.go

```diff
@@ -85,6 +85,10 @@ func (q *querier) lvals(qs []storage.Querier, n string) ([]string, storage.Warni
 	return mergeStrings(s1, s2), ws, nil
 }
 
+func (q *querier) NewManager() storage.SelectManager {
```

Comment:
Why does this require changes to the tsdb?

This should be built on top of the tsdb.

Reply:
I'm aware that you don't want any changes to tsdb, from the previous PRs, for this feature. However, I couldn't find a clean way to implement this without changing the `storage.Querier` interface.

So I had to introduce a minor change to the `storage.Querier` interface, hence the tsdb changes. I tried to keep the changes as few as possible.